vpr/src/place/initial_placement.cpp

```diff
@@ -577,9 +641,8 @@ static bool try_centroid_placement(const t_pl_macro& pl_macro,
 
     //centroid suggestion was either occupied or does not match block type
     //try to find a near location that meet these requirements
-    bool neighbor_legal_loc = false;
-    if (!is_loc_legal(centroid_loc, pr, block_type)) {
-        neighbor_legal_loc = find_centroid_neighbor(centroid_loc, block_type, false, blk_loc_registry, rng);
+    if (!is_loc_legal(centroid_loc, pr, block_type) || try_neighbour_due_to_subtile) {
```

Comment:
I think you can simplify this to if (!found_legal_subtile)    // pick whatever variable name makes sense
if you change the find_subtile_in_location routine to return false if it fails to find a valid subtile at the given location for *any* reason.

It makes that routine more clear, and the logic here more straightforward.